Add recipe for llm-d-fs connector storage offloading with Lustre (#1427)
llm-d/llm-d: guides/tiered-prefix-cache/storage/README.md

| KVCache > HBM + CPU RAM | Mean TTFT (second) | P90 TTFT (second) | Mean E2E Latency (second) | P90 E2E Latency (second) | Input Throughput (token per second) | Output Throughput (token per second) | Overall Throughput (token per second) |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| **Baseline vLLM + CPU offloading** | 25.38 | 37.74 | 56.21 | 69.69 | 18607 | 354 | 18962 |
| **vLLM + CPU offloading + Lustre** | 20.12 (-21%) | 34.02 (-9.9%) | 45.83 (-18%) | 58.73 (-16%) | 22827 (+23%) | 435 (+23%) | 23262 (+23%) |

##### 70K system prompt length (KVCache size 1.3TiB GiB) — KV Cache >> (HBM + CPU RAM)

| KVCache >> HBM + CPU RAM | Mean TTFT (second) | P90 TTFT (second) | Mean E2E Latency (second) | P90 E2E Latency (second) | Input Throughput (token per second) | Output Throughput (token per second) | Overall Throughput (token per second) |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| **Baseline vLLM + CPU offloading** | 58.02 | 74.75 | 87.99 | 105.46 | 16598 | 226.65 | 16825 |
| **vLLM + CPU offloading + Lustre** | 45 (-22%) | 64.79 (-13%) | 68.28 (-22%) | 87.47 (-17%) | 21364 (+28.71%) | 291 (+28.39%) | 21656 (+28.71%) |

> #### **LLM-D FS Connector Configuration:**

Reviewer: I think we should split this: keep the original file focused on storage only, and add a separate section (or file) for the multi-connector setup.

Reviewer 2: @kfirtoledo Do you think there is a use case where you want only storage offloading but no CPU offloading? I cannot think of any. So I think we should enable CPU offloading by default, this also aligns with the "tired prefix cache" guide name.

Author: Added the multi-connector to base as per suggestion. Please review

>   * The LLM-D FS connector does not currently support multi-tiered offloading (e.g., offloading to both CPU and storage backends). To enable a tiered structure, we used a Multi-connector combining a CPU Offloading connector with a storage-backed native offloading connector.

Reviewer: ```suggestion
  * The offloading connector supports multi-tiered offloading (e.g., offloading to both CPU and storage backends) through a Multi-connector setup, which combines an Offloading connector with a CPU backend and an Offloading connector with an LLM-D FS backend.
```
